### PR TITLE
[ansible-test] Add Fedora 33

### PR DESCRIPTION
**Pull request by relrod**
##### SUMMARY

Change:
- Changes to make F33 work with current tests
- Add F33 to docker.txt
- Add F33 to AZP config

Test Plan:
- ci_complete

Signed-off-by: Rick Elrod <rick@elrod.me>

##### ISSUE TYPE

- Test Pull Request

##### COMPONENT NAME

ansible-test

### CODE REVIEW

File: test/lib/ansible_test/_data/completion/docker.txt
Comment (webknjaz) [inline test/lib/ansible_test/_data/completion/docker.txt]: Is it possible to also make use of the 3.10 alpha that's available in F33? There seems to be demand for that (https://github.com/ansible/ansible/pull/72852).

**Comment by bek24** [comment]
Looks wrong !!

File: .azure-pipelines/azure-pipelines.yml
Comment (mattclay) [inline .azure-pipelines/azure-pipelines.yml:124]: We only keep the latest two Fedora releases, so we should drop Fedora 31 now.

File: test/lib/ansible_test/_data/completion/docker.txt
Comment (mattclay) [reply test/lib/ansible_test/_data/completion/docker.txt]: Support for 3.10 is something that should be worked on separately from this PR.

File: test/lib/ansible_test/_data/requirements/constraints.txt
Comment (mattclay) [inline test/lib/ansible_test/_data/requirements/constraints.txt]: I'd make the changes for pylint in a separate PR, since they're likely going to be backported differently than the other changes in this PR.

**Review by mattclay — CHANGES_REQUESTED** [review]

File: test/lib/ansible_test/_data/completion/docker.txt
Comment (webknjaz) [reply test/lib/ansible_test/_data/completion/docker.txt]: Totally, I just wanted to raise attention to that PR.

File: test/lib/ansible_test/_data/requirements/constraints.txt
Comment (relrod) [reply test/lib/ansible_test/_data/requirements/constraints.txt]: I can do it in a separate commit in the same PR, but doing a separate PR will be hard since they're needed for tests to pass ;)

**Comment by relrod** [comment]
# NOTE:

If someone merges this before I do -- please keep the commits intact, don't squash them.

File: test/lib/ansible_test/_data/requirements/constraints.txt
Comment (mattclay) [reply test/lib/ansible_test/_data/requirements/constraints.txt]: I'd like to see the pylint changes in a separate PR that we merge first, then re-run tests on this PR before merging.

**Comment by mattclay** [comment]
This is currently blocked on a separate PR (https://github.com/ansible/ansible/pull/72972) which avoids pylint test errors on Python 3.9.

File: changelogs/fragments/ansible-test-fedora33.yml
Comment (mattclay) [inline changelogs/fragments/ansible-test-fedora33.yml]: Although we're removing it from CI, it's not being removed from `ansible-test`:

```suggestion
```

**Review by mattclay — CHANGES_REQUESTED** [review]

**Review by mattclay — APPROVED** [review]

**Pull request merged**
